### PR TITLE
Recurring Transaction Feature

### PR DESCRIPTION
**Pull request by GilgameshTC**
- [x] Saving recurring transactions (eager)

- [x] Deleting all instances of a recurring transaction

- [x] Update recurring transaction (repeats)

### CODE REVIEW

**Comment by GilgameshTC** [comment]
Question : If we update a recurring instance to have an extra tag, do we want to see this change across all instances?

**Comment by GilgameshTC** [comment]
Implement TransactionManager test case:
- [x] saveRecurringTransaction()
- [x] updateRecurringTransaction()
- [x] deleteAllRecurringInstances()

File: bacon/controller/CoreLogic.swift [hunk 1]
```diff
@@ -51,6 +51,15 @@ class CoreLogic: CoreLogicInterface {
         try transactionManager.saveTransaction(currentTransaction)
     }
 
+    func deleteAllRecurringInstances(of transaction: Transaction) throws {
+        guard transaction.frequency.nature == .recurring else {
+            throw InvalidArgumentError(message: """
+                deletellRecurringInstances() requires transaction to be recurring.
```
Comment (fterh) [inline bacon/controller/CoreLogic.swift]: Typo

File: bacon/controller/TransactionManager.swift [hunk 2]
```diff
@@ -70,12 +74,102 @@ class TransactionManager: Observer {
...
+        return recurringTransactions
+    }
+
+    /// An edited recurring transaction should have its changes
+    /// apply across all recurring instances.
+    /// We disallow editing the date of a recurring transaction as
+    /// users are able to edit a recurring transaction at an arbitrary instance of it,
+    /// if we allow user to edit the date, we will be unable to back track and
+    /// find out which instance the transaction was edited at.
+    func updateRecurringTransaction(_ transaction: Transaction) throws {
```
Comment (fterh) [inline bacon/controller/TransactionManager.swift:147]: Maybe you can put a requires clause to require the date not be modified

**Review by fterh — APPROVED** [review]

**Pull request merged**